PolyEngine/Engine/Src/DebugDrawComponents.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include "ComponentBase.hpp"
+#include "DebugConfig.hpp"
+#include "Dynarray.hpp"
+#include "Vector.hpp"
+
+namespace Poly
+{
+	/// Component used for labeling entities for which the debug primitives 
+	/// will be drawn.
+	class ENGINE_DLLEXPORT DebugDrawableComponent : public ComponentBase
+	{
+	public:
+		DebugDrawableComponent(DebugDrawPreset preset) : entityPreset(preset) { };
+
+		const DebugDrawPreset entityPreset;
+	};
+
+	REGISTER_COMPONENT(ComponentsIDGroup, DebugDrawableComponent)
+
+	class ENGINE_DLLEXPORT DebugDrawLinesComponent : public ComponentBase
```

Comment:
Then this one component would store all data (multiple `DynArrays`) for points, lines, etc. Wouldn't it make the component too heavyweight? IDK maybe my fear of large components is irrational.
But there is also something else; `DebugDrawWorldComponent` IMO is to general. A division for smaller, more specific components is more descriptive.

Reply:
Well, there is nothing bad with heavy components as long as they contain data that is all part of certain functionality. For me `DebugDrawWorldComponent` is appropriate because this is component that contains data used by `DebugDrawSystem`. If you have any other propositions, then feel free to share them.

Reply:
It's true that points, lines, etc. being 'debug vertex data' have common denominator but they still have to be handled differently in rendering pass (they are loaded on different buffers, drawn by another draw call, might use different shaders).

Reply:
Well that logic should be handled by RenderingDevice, not Engine.